liblangutil/Scanner.h

```diff
@@ -98,6 +98,7 @@ class Scanner
 	std::string const& source() const noexcept { return m_source->source(); }
 
 	std::shared_ptr<CharStream> charStream() noexcept { return m_source; }
+	std::shared_ptr<CharStream> const& constCharStream() const noexcept { return m_source; }
```

Comment:
Shouldn't it be
```suggestion
	std::shared_ptr<CharStream const> charStream() const noexcept { return m_source; }
```
instead?